client/security/src/main/java/org/eclipse/kapua/client/security/amqpclient/Client.java

```diff
@@ -73,12 +77,61 @@ public TextMessage createTextMessage() throws JMSException {
         return session.createTextMessage();
     }
 
-    public boolean isConnected() {
+    public void stop() throws JMSException {
+        active = false;
+        disconnect();
+    }
+
+    private void disconnect() throws JMSException {
+        if (connection != null) {
+            connection.close();
```

Comment:
Shouldn't we set the connectionStatus here to false?

Reply:
I was relying on the ExceptionListener but you are right: it's better to set it here.

Reply:
Okay let's add it in, then I'd be okay approving this.